server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/cassandra/CassandraBlobExistenceTesterDAO.java

```diff
@@ -74,7 +74,8 @@ private PreparedStatement prepareSelect(Session session) {
     private PreparedStatement prepareSelectBucket(Session session) {
         return session.prepare(select()
             .from(TABLE_NAME)
-            .where(eq(BUCKET_NAME, bindMarker(BUCKET_NAME))));
+            .where(eq(BUCKET_NAME, bindMarker(BUCKET_NAME)))
+            .allowFiltering());
```

Comment:
I think that such a decision should always comes with a comment

Reply:
:+1: we need to add a comment.

https://github.com/linagora/james-project/pull/2999#issuecomment-566964920 provides the explanation.

We decided to allowFiltering to reduce streaming between Cassandra and James as Cassandra would end up reading all entries anyway. allowFiltering is thus a better option than a client table full scan as it saves some network.